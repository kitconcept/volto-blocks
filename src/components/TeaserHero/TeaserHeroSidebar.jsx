import React from 'react';
import { Segment } from 'semantic-ui-react';
import { FormattedMessage } from 'react-intl';

import TeaserHeroData from './TeaserHeroData';

const TeaserHeroSidebar = props => {
  const { sidebarData } = props;

  const teaserHeroSidebarDataRender = sidebarData ? (
    <>{sidebarData(props)}</>
  ) : (
    <TeaserHeroData {...props} />
  );

  return (
    <Segment.Group raised>
      <header className="header pulled">
        <h2>
          <FormattedMessage id="Teaser Hero" defaultMessage="Teaser Hero" />
        </h2>
      </header>

      {teaserHeroSidebarDataRender}
    </Segment.Group>
  );
};

export default TeaserHeroSidebar;
