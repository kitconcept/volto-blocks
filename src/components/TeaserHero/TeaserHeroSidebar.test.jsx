import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import TeaserHeroSidebar from './TeaserHeroSidebar';

const mockStore = configureStore();

test('renders a Teaser Hero Sidebar component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <TeaserHeroSidebar
        id="1c58b5a2-26fb-4112-832b-fad88af1ae0e"
        data={{
          '@type': 'teaserHero',
          href: '/de/a-page/news1',
          variation: 'left',
          padded: true,
        }}
        block="1c58b5a2-26fb-4112-832b-fad88af1ae0e"
        onChangeBlock={() => {}}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
