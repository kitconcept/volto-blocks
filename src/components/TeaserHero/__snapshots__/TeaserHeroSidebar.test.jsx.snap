// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Teaser Hero Sidebar component 1`] = `
<div
  className="ui raised segments"
>
  <header
    className="header pulled"
  >
    <h2>
      <span>
        Teaser Hero
      </span>
    </h2>
  </header>
  <div
    className="ui segment form sidebar-image-data"
  >
    <div
      className="inline field"
      id="field-source"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-source"
              >
                Source
              </label>
            </div>
          </div>
          <div
            className="eight wide column"
          >
            <div
              className="ui icon input"
            >
              <input
                disabled={false}
                id="field-source"
                name="source"
                onChange={[Function]}
                type="text"
                value="/de/a-page/news1"
              />
              <i
                aria-hidden="true"
                attributes={
                  Object {
                    "viewBox": "",
                    "xmlns": "",
                  }
                }
                className="icon"
              />
            </div>
            <button
              onClick={[Function]}
            >
              <svg
                className="icon"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
                onClick={null}
                style={
                  Object {
                    "fill": "currentColor",
                    "height": "18px",
                    "width": "auto",
                  }
                }
                viewBox=""
                xmlns=""
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="inline field"
      id="field-openLinkInNewTab"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="twelve wide column"
          >
            <div
              className="wrapper"
            >
              <div
                className="ui fitted checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  disabled={false}
                  name="field-openLinkInNewTab"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label />
              </div>
              <label
                className="ui basic horizontal label"
                htmlFor="field-openLinkInNewTab"
                onClick={[Function]}
              >
                Open in a new tab
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui segment form sidebar-image-data"
  >
    <div
      className="inline field"
    >
      <div
        className="ui grid"
      >
        <div
          className="row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-align"
              >
                <span>
                  Variation
                </span>
              </label>
            </div>
          </div>
          <div
            className="eight wide column align-tools"
          >
            <div>
              <div
                className="ui buttons"
              >
                <button
                  aria-label="Top"
                  className="ui basic icon button"
                  onClick={[Function]}
                >
                  <svg
                    className="icon"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": undefined,
                      }
                    }
                    onClick={null}
                    style={
                      Object {
                        "fill": "currentColor",
                        "height": "24px",
                        "width": "auto",
                      }
                    }
                    viewBox=""
                    xmlns=""
                  />
                </button>
              </div>
              <div
                className="ui buttons"
              >
                <button
                  aria-label="Left"
                  className="ui active basic icon button"
                  onClick={[Function]}
                >
                  <svg
                    className="icon"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": undefined,
                      }
                    }
                    onClick={null}
                    style={
                      Object {
                        "fill": "currentColor",
                        "height": "24px",
                        "width": "auto",
                      }
                    }
                    viewBox=""
                    xmlns=""
                  />
                </button>
              </div>
              <div
                className="ui buttons"
              >
                <button
                  aria-label="Right"
                  className="ui basic icon button"
                  onClick={[Function]}
                >
                  <svg
                    className="icon"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": undefined,
                      }
                    }
                    onClick={null}
                    style={
                      Object {
                        "fill": "currentColor",
                        "height": "24px",
                        "width": "auto",
                      }
                    }
                    viewBox=""
                    xmlns=""
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="inline field"
      id="field-isPadded"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="twelve wide column"
          >
            <div
              className="wrapper"
            >
              <div
                className="ui fitted checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  disabled={false}
                  name="field-isPadded"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label />
              </div>
              <label
                className="ui basic horizontal label"
                htmlFor="field-isPadded"
                onClick={[Function]}
              >
                Add padding
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
